animations/lib/src/basics/08_animated_positioned.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class AnimatedPositionedDemo extends StatefulWidget {
+  static String routeName = '/basics/08_animated_positioned';
+
+  _AnimatedPositionedDemoState createState() => _AnimatedPositionedDemoState();
+}
+
+class _AnimatedPositionedDemoState extends State<AnimatedPositionedDemo> {
+  double topPosition;
+  double rightPosition;
+
+  double generateTopPosition(double top) => Random().nextDouble() * top;
+
+  double generateRightPosition(double right) => Random().nextDouble() * right;
+
+  void initState() {
+    super.initState();
+    topPosition = generateTopPosition(30);
+    rightPosition = generateRightPosition(30);
+  }
+
+  void changePosition(double top, double right) {
+    setState(() {
+      topPosition = generateTopPosition(top);
+      rightPosition = generateRightPosition(right);
+      print(topPosition);
+      print(rightPosition);
+    });
+  }
+
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+    // AnimatedPositioned animates position of the given child over the given duration of time.
+    return Scaffold(
+      appBar: AppBar(),
+      body: Container(
+        height: size.height,
+        width: size.width,
+        padding: EdgeInsets.all(8),
+        child: Stack(
+          children: <Widget>[
+            AnimatedPositioned(
+              top: topPosition,
+              right: rightPosition,
+              duration: Duration(seconds: 1),
+              child: RaisedButton(
+                color: Colors.purple,
+                child: Text(
+                  'Change Position',
+                  style: TextStyle(
+                    color: Colors.white,
+                  ),
+                ),
+                onPressed: () =>
+                    changePosition(size.height / 2, size.width / 2),
```

Comment:
It looks like you've run into a tricky bit here, since it's difficult to know the size of the button in advance and adjust your position values accordingly, so you're just using half the screen and hoping everything fits (try upping the text size in your device's settings -- you may be surprised!).

Normally I'd suggest using an Align instead, but that would make this into an AnimatedAlign sample rather than an AnimatedPosition one.

There are a couple things you can do, though. One would be to wrap the Button into a FittedBox with a fixed width and height. That would give you a value you can count on.

Reply:
Using `FittedBox` and calculating axis position will still overflow the `RaisedButton` from any one side. If I define `size.width - 100`. The box may overflow from left side.

Reply:
I guess that will not work. I tried wrapping the button with a `Container` having fixed height and width, but it will still overflow the `RaisedButton` from any side.

Reply:
I guess this won't work. I tried wrapping `RaisedButton` in a `Container` having fixed height and width. For example
Suppose, I have width = 500 and my `Container` width is 100, If I subtract 500-100 to generate a random number and give that to right parameter in `AnimatedPositioned` then 
it may be a case that it will overflow from the left side.

Reply:
If you use a FittedBox widget, you should be able to specify the exact dimensions you want, and FittedBox will scale the Button to make it fit.

Another potential solution would be to just use a container with a fixed height and width, and remove the button entirely. If it's the only thing onscreen, it'll be pretty clear to users that they should tap on it.